var/spack/repos/builtin/packages/tk/package.py

```diff
@@ -38,18 +38,16 @@ class Tk(AutotoolsPackage):
     version('8.6.5', '11dbbd425c3e0201f20d6a51482ce6c4')
     version('8.6.3', '85ca4dbf4dcc19777fd456f6ee5d0221')
 
-    variant('X', default=False, description='Enable X11 support')
-
-    depends_on("tcl")
-    depends_on("libx11", when='+X')
```

Comment:
Summary of #3047:

If you don't have X11 libraries on your system and you try to build `tk~X`, it crashes. There is a `--without-x` flag, but even if you use it, you see:
```
fatal error: X11/Xlib.h: No such file or directory
 #   include <X11/Xlib.h>
                         ^
compilation terminated.
make: *** [tkStubLib.o] Error 1
```
Theoretically, we might be able to build with Aqua instead of X11, but one or the other seems to be required. Since I don't know or care about Aqua, I decided to go with X11 all the time. If someone needs Aqua instead, they can hack this. Homebrew makes it optional:
https://github.com/Homebrew/homebrew-dupes/blob/master/tcl-tk.rb#L66